var/spack/repos/spack_repo/builtin/packages/namd/package.py

```diff
@@ -220,9 +224,13 @@ def _edit_arch_generic(self, spec, prefix):
                             "NAMD_ARCH = {0}".format(self.arch),
                             "CHARMARCH = {0}".format(self.spec["charmpp"].charmarch),
                             "CXX = {0.cxx} {0.cxx11_flag}".format(self.compiler),
-                            "CXXOPTS = {0}".format(optim_opts),
+                            "CXXOPTS = {0} {1}".format(
+                                optim_opts, " ".join(spec.compiler_flags["cxxflags"])
```

Comment:
Is spec.compiler_flags documented somewhere? Where does it come from? Is it redundant with the optims_opts flags that are already set or self.compiler.cxx?

Reply:
Unfortunately there is no good documentation about it.
In [Packaging guide](https://spack.readthedocs.io/en/latest/packaging_guide.html#compiler-flags) it is recommended to use `flag_handler`. Since no flag_handler was provided, I think this method of handling compiler flags is a good alternative.

There are many other packages that use `spec.compiler_flags`: esmf, ascent, mfem, netlib_lapack, etc.

IIt’s not redundant because in the current setup, the flags I pass through Spack (using cflags or cxxflags) won’t end up in `CXXOPTS` or `COPTS`.

`self.compiler.cxx` sets only compiler.
`optim_opts`is a minimal set of flags that ensures good performance (can't be extended with current approach).

Reply:
Good point!